### ISSUE TITLE
Topic assigment invalid after subscription

### ISSUE THREAD

**Issue by martin-intex** [dpkp/kafka-python#1981]
When using the KafkaConsumer with the subscription method a call directly after the subscription (i.e. before any message was received) to the assignment() method returns an empty set.
Only after a message was received the currently assigned partition can be examined. This knowledge however would be required to reset the offset of that consumer group. (Or is there is another way to reset the consumer group offset?)

For example:
```
consumer = KafkaConsumer(bootstrap_servers='localhost:9092',
                     auto_offset_reset='earliest', 
                     consumer_timeout_ms=1000) 
consumer.subscribe(['my-topic'] 
print(consumer..assignment()}) 
```
Would result in `set()`

**Comment by romulo-dbts** [comment]
I am not sure if this is what you want to do, but take a look on both methods `kafka.consumer.group.KafkaConsumer.seek` (https://github.com/dpkp/kafka-python/blob/master/kafka/consumer/group.py#L795) and `kafka.consumer.group.KafkaConsumer.seek_to_beginning` (https://github.com/dpkp/kafka-python/blob/master/kafka/consumer/group.py#L823). They might help you to start consuming from a particular offset.

**Comment by martin-intex** [comment]
Thank you romulo for your reply. You are right that I like to use the two methods. However - both require a partition argument. So my plan was to subscribe to a topic based on my consumer group, see which partition I got assigned from the zookeeper and set the offset accordingly. 
Unfortunately I can not get the information which partition I am assigned to until I got one message. So I thought it might be a bug that the consumer..assignment() does not return any data after successful subscribing but before having received any message.
